pkg/controller/core/workload_controller.go

```diff
@@ -184,7 +184,7 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		}
 	} else {
 		var updated, evicted bool
-		reason := kueue.WorkloadEvictedByDeactivation
+		reason := kueue.WorkloadDeactivated
```

Comment:
As we are changing this anyway, maybe we could also adjust `reason += dtCond.Reason` (couple of lines below), as I mentioned in https://github.com/kubernetes-sigs/kueue/issues/3512. 

Because currently we have reasons for `Evicted` like `DeactivatedRequeuingLimitExceeded` or `DeactivatedAdmissionCheck`. I think they would read better if we use `DeactivatedDueToRequeuingLimitExceeded` and `DeactivatedDueToAdmissionCheck`. 

I consider combining the changes to avoid changing the reasons in consecutive releases.

Any thoughts @tenzen-y @PBundyra ?

Reply:
Sgtm

Reply:
Done